scheduler/reconcile.go

```diff
@@ -218,23 +218,21 @@ func (a *allocReconciler) Compute() *reconcileResults {
 
 	// Mark the deployment as complete if possible
 	if a.deployment != nil && complete {
-
-		var status string
-		var desc string
-
 		if a.job.IsMultiregion() {
-			status = structs.DeploymentStatusBlocked
-			desc = structs.DeploymentStatusDescriptionBlocked
+			if a.deployment.Status != structs.DeploymentStatusUnblocking {
```

Comment:
for my edification - do we need to worry about other statues?  Is it possible here to have a race causing this to be successful (state after unblocking?!)?

Reply:
Good point... if the deploymentwatcher has already checkpointed this as successful we'd update it unnecessarily. That wouldn't result in the error we were seeing in the deploymentwatcher (because by that point the deploymentwatcher has exited), but for robustness it's better to handle this case properly. Will fix.

Reply:
Done in 24edd7a